lldb/source/Plugins/ExpressionParser/Swift/SwiftExpressionParser.cpp

```diff
@@ -640,36 +657,42 @@ static void AddRequiredAliases(Block *block, lldb::StackFrameSP &stack_frame_sp,
     swift::BoundGenericEnumType *optional_type =
         referent_type->getAs<swift::BoundGenericEnumType>();
 
-    if (!optional_type)
-      return;
+    if (!optional_type || optional_type->getGenericArgs().empty())
+      return Status("Could not add required aliases: could not get optional "
+                    "type for self.");
 
     swift::Type first_arg_type = optional_type->getGenericArgs()[0];
 
-    swift::ClassType *self_class_type =
-        first_arg_type->getAs<swift::ClassType>();
-
-    if (!self_class_type)
-      return;
+    // In Swift only class types can be weakly captured.
+    if (!llvm::isa<swift::ClassType>(first_arg_type) &&
+        !llvm::isa<swift::BoundGenericClassType>(first_arg_type))
+      return Status("Unable to add the aliases the expression needs because "
+                    "weakly captured type is not a class type.");
 
-    imported_self_type = ToCompilerType(self_class_type);
+    imported_self_type = ToCompilerType(first_arg_type);
   }
 
   imported_self_type_flags.Reset(imported_self_type.GetTypeInfo());
-  if (imported_self_type_flags.AllClear(lldb::eTypeIsGenericTypeParam)) {
-    swift::ValueDecl *type_alias_decl = nullptr;
-
-    type_alias_decl = manipulator.MakeTypealias(
-        swift_ast_context.GetASTContext()->getIdentifier("$__lldb_context"),
-        imported_self_type);
-
-    if (!type_alias_decl)
-      LLDB_LOG(GetLog(LLDBLog::Expressions),
-               "SEP:AddRequiredAliases: Failed to make the $__lldb_context "
-               "typealias.");
-  } else
+  if (imported_self_type_flags.Test(lldb::eTypeIsGenericTypeParam)) {
     LLDB_LOG(GetLog(LLDBLog::Expressions),
              "SEP:AddRequiredAliases: Failed to resolve the self archetype - "
              "could not make the $__lldb_context typealias.");
+    return Status("Could not add required aliases: failed to resolve the self "
+                  "archetype.");
+  }
+  swift::ValueDecl *type_alias_decl = manipulator.MakeTypealias(
+      swift_ast_context.GetASTContext()->getIdentifier("$__lldb_context"),
+      imported_self_type);
+
+  if (!type_alias_decl) {
+    LLDB_LOG(GetLog(LLDBLog::Expressions),
+             "SEP:AddRequiredAliases: Failed to make the $__lldb_context "
+             "typealias.");
+    return Status("Could not add required aliases: could not create lldb "
```

Comment:
I missed this one too.

```suggestion
    return Status("Unable to add the aliases the expression needs because Swift expression parser "
```